src/lighteval/models/endpoints/inference_providers_model.py

```diff
@@ -0,0 +1,273 @@
+            model_name=config.model,
+            model_sha="",
+            model_dtype=None,
+            model_size="",
+        )
+        self.model = config.model
+        self.provider = config.provider
+        self.generation_parameters = config.generation_parameters
+
+        self.API_MAX_RETRY = 5
+        self.API_RETRY_SLEEP = 3
+        self.API_RETRY_MULTIPLIER = 2
+        self.pairwise_tokenization = False
+        self.semaphore = asyncio.Semaphore(config.parallel_calls_count)  # Limit concurrent API calls
+
+        self.client = AsyncInferenceClient(
+            provider=self.provider,
+            timeout=config.timeout,
+            proxies=config.proxies,
+        )
+        self._tokenizer = AutoTokenizer.from_pretrained(self.model)
+
+    def _encode(self, text: str) -> dict:
+        enc = self._tokenizer(text=text)
+        return enc
+
+    def tok_encode(self, text: str | list[str]):
+        if isinstance(text, list):
+            toks = [self._encode(t["content"]) for t in text]
+            toks = [tok for tok in toks if tok]
+            return toks
+        return self._encode(text)
+
+    async def __call_api(self, prompt: List[dict], num_samples: int) -> Optional[ChatCompletionOutput]:
+        """Make API call with exponential backoff retry logic.
+
+        Args:
+            prompt: List of message dictionaries for chat completion
+            num_samples: Number of completions to generate
+
+        Returns:
+            API response or None if all retries failed
+        """
+        for attempt in range(self.API_MAX_RETRY):
+            try:
+                kwargs = {
+                    "model": self.model,
+                    "messages": prompt,
+                    "n": num_samples,
+                }
+                kwargs.update(self.generation_parameters.to_inference_providers_dict())
+                response: ChatCompletionOutput = await self.client.chat.completions.create(**kwargs)
+                return response
+            except Exception as e:
+                wait_time = min(64, self.API_RETRY_SLEEP * (2**attempt))  # Exponential backoff with max 64s
+                logger.warning(
+                    f"Error in API call: {e}, waiting {wait_time} seconds before retry {attempt + 1}/{self.API_MAX_RETRY}"
+                )
+                await asyncio.sleep(wait_time)
+
+        logger.error(f"API call failed after {self.API_MAX_RETRY} attempts, returning empty response.")
+        return None
+
+    async def __call_api_parallel(
+        self,
+        prompts,
+        num_samples: int | list[int],
+    ):
+        results = []
+
+        num_sampless = [num_samples for _ in prompts] if not isinstance(num_samples, list) else num_samples
+        assert (
+            len(prompts) == len(num_sampless)
+        ), f"Length of prompts, return_logitss, max_new_tokenss, num_sampless, stop_sequences, system_prompts should be the same but are {len(prompts)}, {len(num_sampless)}"
```

Comment:
You're not testing all of that are, I think this error message should be adapted